### ISSUE TITLE
Calendar `defaultcalendar` suddenly replaced with `personal`, events gone.

### ISSUE THREAD

**Issue by noniq** [owncloud/core#26315]
Today I noticed that all the events from my personal calendar had disappeared.

TL;DR: It seems that the calendar names `defaultcalendar` (that’s the name in the CalDAV sharing URL) suddenly got deleted and replaced by a new empty calendar named `personal`.

I could re-import all my events from an backup, but I’d really love to understand what had happened here (to make sure it won’t happen again in the future).

Here’s some background: I’m running ownCloud 9.1.0 on an Ubuntu 12.04 server. I almost exlusively use the calendar via CalDAV (macOS, iOS). Looking through backup versions of the `~/Library/Calendars/<some-uuid>.caldav` folder, where the macOS calendar app caches the calendar data of this caldav account, I noticed that sometime between yesterday and today noon the subfolder for the old “defaultcalendar” disappeared and instead a new subfolder for the new “personal” calendar started to show up.

Of course I can not completely eliminate the possibilty that I somehow deleted the original calendar on one of my devices per accident, but even if this was the case, where would the new calendar “personal” have come from? (I surely did not create this new calendar. Does owncloud automatically create a new default calendar when the previous one gets deleted?).
 
The owncloud logfile doesn’t seem to contain anything relevant (I’m not really sure what to look for, though). Grepping for `defaultcalendar` only shows a handful of messages like this (all older than 3 weeks):

```
{"reqId":"EOTCPFKGjwW9+DOhHP5A","remoteAddr":"80.110.47.228","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/3rdparty\/sabre\/http\/lib\/Sapi.php:83) at \/var\/www\/owncloud\/3rdparty\/sabre\/http\/lib\/Sapi.php#63","level":3,"time":"2016-09-20T10:00:27+00:00","method":"GET","url":"\/remote.php\/dav\/calendars\/stefan\/defaultcalendar\/","user":"--"}
```

And grepping for `personal` gives me a bunch of these messages (starting around the time where I started to try to add new events to this new `personal` calendar`):
```
{"reqId":"2V7PsZkeVfAS93BX6Pam","remoteAddr":"80.109.59.57","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 401 No 'Authorization: Basic' header found. Either the client didn't send one, or the server is mis-configured, No 'Authorization: Basic' header found. Either the client didn't send one, or the server is mis-configured\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotAuthenticated\",\"Code\":0,\"Trace\":\"#0 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#2 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(446): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#3 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Server.php(169): Sabre\\\\DAV\\\\Server->exec()\\n#5 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(29): OCA\\\\DAV\\\\Server->exec()\\n#6 \\\/var\\\/www\\\/owncloud\\\/remote.php(164): require_once('\\\/var\\\/www\\\/ownclo...')\\n#7 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php\",\"Line\":188,\"User\":false}","level":0,"time":"2016-10-08T14:45:00+00:00","method":"PUT","url":"\/remote.php\/dav\/calendars\/stefan\/personal\/0018F01B-051F-4F02-9B54-4EBB228CA304.ics","user":"--"}
```
(By the way, is it safe to ignore these errors? Adding events via CalDAV seems to work fine, also syncing between all my devices seems to work normally.)

Any pointers where I could look to get a better post mortem and deeper understanding of what happened?

**Comment by PVince81** [comment]
@DeepDiver1975

**Comment by DeepDiver1975** [comment]
A calendar named 'personal' is created for a user on login in case no other calendar exists.

So as far as I can think about this issue .... I'd say call calendars got deleted and upon login a new, empty calendar was created.

You can analyse this by having a look at your access log of the web server - there should be a DELETE request for the defaultcalendar url.

**Comment by noniq** [comment]
Thanks for the tip! There’s indeed an appropriate DELETE request in the access log, seemingly triggered from my iPhone. No idea how that happened, but now I’m at least sure that it’s no bug with owncloud :-)

Should I open a separate issue for these "HTTP/1.1 401 No 'Authorization: Basic' header found." log messages? (I couldn’t find an existing one.) Or are these safe to ignore?

**Comment by PVince81** [comment]
Safe to ignore, they are part of the authorization protocol
